client/lib/analytics/test/google-analytics-4.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @jest-environment jsdom
+ */
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import { TRACKING_IDS } from '../ad-tracking/constants';
+import { setup, firePageView, Ga4PropertyGtag } from '../ad-tracking/google-analytics-4';
+
+jest.mock( 'calypso/lib/jetpack/is-jetpack-cloud' );
+
+describe( 'Google Analytics 4 implementation', () => {
+	beforeEach( () => {
+		jest.spyOn( window, 'gtag' ).mockImplementation( () => [] );
+		// window.gtag = jest.fn( () => [] );
+	} );
+
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	describe( 'Google Analytics 4 initialization on Jetpack Cloud and WPcom', () => {
+		test( 'Fire Jetpack Gtag on Jetpack Cloud', () => {
+			// We're on Jetpack Cloud.
+			isJetpackCloud.mockReturnValue( true );
+			const params = {};
+
+			setup( params );
+			expect( window.gtag ).toHaveBeenCalledWith(
+				'config',
+				TRACKING_IDS.jetpackGoogleGA4Gtag,
+				params
+			);
+			// Should also init WPcom gtag.
+			expect( window.gtag ).toHaveBeenCalledWith(
+				'config',
+				TRACKING_IDS.wpcomGoogleGA4Gtag,
+				params
+			);
+		} );
+		test( 'Do not fire Jetpack Gtag on WPcom', () => {
```

Comment:
I understood the sense of this test as one checking if we do NOT fire Jetpack Gtag rather than just checking that WPcom has been fired.

I would add the assertion:
```
expect( window.gtag ).not.toHaveBeenCalledWith(
 'config',
 TRACKING_IDS.jetpackGoogleGA4Gtag,
 params
);
```

What do you think?

Reply:
That's a good idea. I've added that assertion as well now. I will work more on test coverage of the lib, since we'll do more refactors -- and we can do those more confidently with more tests. :)